### PR TITLE
Feature/comet logging

### PR DESCRIPTION
**Pull request by sherpan**
# Description

Add comet as a potential logger for the repo

## Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which refactors the code base)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the [pre-commit style and check guidelines](https://openvinotoolkit.github.io/anomalib/guides/using_pre_commit.html#pre-commit-hooks) of this project.
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Comment by ashwinvaidya17** [comment]
@sherpan can you also have a look at https://github.com/openvinotoolkit/anomalib/runs/8065761026?check_suite_focus=true

File: anomalib/utils/callbacks/graph.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def on_train_end(self, trainer: Trainer, pl_module: LightningModule) -> None:
         """
 
         for logger in trainer.loggers:
-            if isinstance(logger, AnomalibTensorBoardLogger):
+            if isinstance(logger, AnomalibTensorBoardLogger) or isinstance(logger, AnomalibCometLogger):
```
Comment (samet-akcay) [inline anomalib/utils/callbacks/graph.py:41]: As suggested by the codacy, we could do something like 

```
if isinstance(logger, (AnomalibCometLogger, AnomalibTensorBoardLogger)):
```

File: docs/blog/001-train-custom-dataset/README.md [hunk 2]
```diff
@@ -176,14 +176,21 @@ results
...
+To use TensorBoard and/or W&B logger and/or Comet logger, ensure that the logger parameter is set to `comet`, `tensorboard`, `wandb` or `[tensorboard, wandb]` in the `logging` section of the config file.
 
-An example configuration for saving to TensorBoard is shown in the figure below. Similarly after setting logger to `wandb` you will see the images on your wandb project dashboard.
+An example configuration for saving to TensorBoard is shown in the figure below. Similarly after setting logger to `wandb` or 'comet' you will see the images on your wandb and/or comet project dashboard.
 
 ```yaml
-logging:
-  log_images_to: [tensorboard]
-  logger: tensorboard # options: [tensorboard, wandb, csv] or combinations.
+  visualization:
```
Comment (samet-akcay) [inline docs/blog/001-train-custom-dataset/README.md:184]: I guess `visualization` should not be indented.

File: anomalib/utils/loggers/comet.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+from pytorch_lightning.loggers.comet import CometLogger
+from pytorch_lightning.utilities import rank_zero_only
+
+
+from .base import ImageLoggerBase
+
+
+class AnomalibCometLogger(ImageLoggerBase, CometLogger):
+    """Logger for comet. 
+
```
Comment (samet-akcay) [inline anomalib/utils/loggers/comet.py:19]: codacy complains about trailing white space in L18 and L19. Would be good to remove them.

**Review by samet-akcay — COMMENTED** [review]
Thanks @sherpan ! I have few minor comments.

File: anomalib/utils/loggers/comet.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+class AnomalibCometLogger(ImageLoggerBase, CometLogger):
+    """Logger for comet.
+    
+    Adds interface for `add_image` in the logger rather than calling the experiment object.
+    
+    Note:
+        Same as the CometLogger provided by PyTorch Lightning and the doc string is reproduced below.
+    
+    Track your parameters, metrics, source code and more using
+    `Comet <https://www.comet.com/?utm_source=pytorch_lightning&utm_medium=referral>`_.
```
Comment (ashwinvaidya17) [inline anomalib/utils/loggers/comet.py:26]: should the utm_source be anomalib?

**Comment by samet-akcay** [comment]
@sherpan, I was wondering if you could install `pre-commit` in your local environment, which would solve all these issues that you are having. Something like 

```
pip install pre-commit
pre-commit run --all-files
```
This would fix the issues, after which you could push your changes.

**Comment by sherpan** [comment]
hey samet, yeah i think you are right! will do that next time? Think i am passing everything now?

**Comment by sherpan** [comment]
confused. here is what i am seeing on my side?
Screen Shot 2022-09-01 at 11 04 26 AM

**Comment by samet-akcay** [comment]
That's peculiar. No worries, though. We'll sort out the rest. Thanks a lot for the contribution! Much appreciated!

**Review by ashwinvaidya17 — APPROVED** [review]
Thanks for the changes

**Review by samet-akcay — APPROVED** [review]

**Review by djdameln — APPROVED** [review]

**Comment by ashwinvaidya17** [comment]
@sherpan There is probably something wrong with your environment. I am getting the same issues.
image
I have pushed the fix to this branch https://github.com/openvinotoolkit/anomalib/tree/fix/ashwin/linting. I cannot update your branch as I don't have permissions for your fork.

**Comment by ashwinvaidya17** [comment]
Or maybe we can merge this branch and I can open a new PR for https://github.com/openvinotoolkit/anomalib/tree/fix/ashwin/linting. @samet-akcay @djdameln ?

**Pull request merged**